modules/contact/test/tests/multiple_contact_pairs/multiple_pairs.i

```diff
@@ -0,0 +1,135 @@
+starting_point = 2e-1
+offset = 1e-2
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+[]
+
+[Mesh]
+  file = multiple_pairs.e
+[]
+
+[Modules/TensorMechanics/Master]
+  [all]
+    add_variables = true
+    strain = FINITE
+    generate_output = 'stress_xx'
+  []
+[]
+
+[Materials]
+  [stiffStuff]
+    type = ComputeIsotropicElasticityTensor
+    block = '1 2 3'
+    youngs_modulus = 1e6
+    poissons_ratio = 0.3
+  []
+  [stiffStuff_stress]
+    type = ComputeFiniteStrainElasticStress
+    block = '1 2 3'
+  []
+[]
+
+[ICs]
+  [disp_y]
+    block = '2 3'
+    variable = disp_y
+    value = '${fparse starting_point + offset}'
+    type = ConstantIC
+  []
+[]
+
+[Contact]
+  [action_name]
+    primary = '20 20'
+    secondary = '10 101'
+    penalty = 1e7
+    formulation = penalty
+    tangential_tolerance = 0.0001
+  []
+[]
+
+[BCs]
+  [botx]
+    type = DirichletBC
+    variable = disp_x
+    preset = false
+    boundary = 40
+    value = 0.0
+  []
+  [boty]
+    type = DirichletBC
+    variable = disp_y
+    preset = false
+    boundary = 40
+    value = 0.0
+  []
+  [topy]
+    type = FunctionDirichletBC
+    variable = disp_y
+    preset = false
+    boundary = '30 301'
+    function = '${starting_point} * cos(2 * pi / 40 * t) + ${offset}'
+  []
+  [leftx]
+    type = FunctionDirichletBC
+    variable = disp_x
+    preset = false
+    boundary = '50 501'
+    function = '1e-2 * t'
+  []
+[]
+
+[Executioner]
+  type = Transient
+  end_time = 100
```

Comment:
total time steps = 50? Would this regression test take too long?

Reply:
It takes a few seconds, there may be a dozen elements. I shortened it to 30 steps. It captures two contact pairs in and out of contact.

Reply:
👍